include/tvm/ir/diagnostic.h

```diff
@@ -37,6 +37,15 @@ namespace tvm {
 using tvm::parser::SourceMap;
 using tvm::runtime::TypedPackedFunc;
 
+/*! \brief The diagnostic level, controls the printing of the message. */
+enum class DiagnosticLevel : int {
+  kBug = 10,
```

Comment:
should we use terminology that reflects logging levels here?

Reply:
+1

Reply:
These are levels for feedback to the user from compiling a relay expression. I'm not sure if it makes sense to merge them as these are user-facing while logging is more for developers.

Reply:
I think logging will be for users when it is possible to adjust the c log level at runtime

Reply:
So you want `LOG(BUG)`, `LOG(ERROR)`, `LOG(WARNING)`, `LOG(NOTE)`, `LOG(HELP)`? Our current levels are `INFO`, `WARNING`, `ERROR`, `FATAL`. We cause alias `INFO` to `NOTE` and then add `BUG` and `HELP`?

Reply:
I actually don't mind these labels for indicating what a help message contains, but it would be helpful to comment the translation to log levels here so I don't have to look that up. I imagine people will try to work with these labels but will be approaching this from "the build log is too spammy"

Reply:
There isn't a translation to log messages here. `DiagnosticLevel` is used within relay and TIR to determine how to print the message (these messages are never hidden or filtered).